engine/src/flutter/impeller/entity/contents/filters/gaussian_blur_filter_contents.cc

```diff
@@ -197,12 +197,36 @@ Scalar FloorToDivisible(Scalar val, Scalar divisor) {
   }
 }
 
+// If `y_coord_scale` < 0.0, the Y coordinate is flipped. This is useful
+// for Impeller graphics backends that use a flipped framebuffer coordinate
+// space.
+//
+// See also: IPRemapCoords in convergions.glsl.
+Rect RemapRectCoords(Rect input, Scalar texture_sampler_y_coord_scale) {
+  if (texture_sampler_y_coord_scale < 0.0f) {
+    return Rect::MakeLTRB(input.GetLeft(), 1.0f - input.GetBottom(),
+                          input.GetRight(), 1.0f - input.GetTop());
+  } else {
+    return input;
+  }
+}
+
 struct DownsamplePassArgs {
   /// The output size of the down-sampling pass.
   ISize subpass_size;
   /// The UVs that will be used for drawing to the down-sampling pass.
   /// This effectively is chopping out a region of the input.
   Quad uvs;
+  /// The bounds used for the downsampling pass of a bounded blur, in the same
+  /// UV space as the texture input of the downsampling pass.
+  ///
+  /// During downsampling, out-of-bound pixels are treated as transparent.
+  std::optional<Rect> downsample_uv_bounds;
+  /// The bounds used for the blurring pass of a bounded blur, in the same UV
+  /// space as the output of the downsampling pass.
+  ///
+  /// During downsampling, out-of-bound pixels are treated as transparent.
+  std::optional<Rect> blur_uv_bounds;
```

Comment:
Can you mention why we track these 2 separately since the input is just one bounds.